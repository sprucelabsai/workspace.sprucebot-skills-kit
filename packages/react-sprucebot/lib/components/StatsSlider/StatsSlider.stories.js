'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _StatsSlider = require('./StatsSlider');

var _StatsSlider2 = _interopRequireDefault(_StatsSlider);

var _StatsSlider3 = require('./StatsSlider.md');

var _StatsSlider4 = _interopRequireDefault(_StatsSlider3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('StatsSlider', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_StatsSlider4.default, function () {
	return _react2.default.createElement(_StatsSlider2.default, {
		stats: [{
			dir: 1,
			value: 12,
			title: 'Things'
		}]
	});
}));