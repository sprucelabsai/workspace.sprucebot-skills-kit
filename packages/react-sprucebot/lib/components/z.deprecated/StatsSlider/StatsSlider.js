"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var StatsSlider =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(StatsSlider, _Component);

  function StatsSlider() {
    (0, _classCallCheck2.default)(this, StatsSlider);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(StatsSlider).apply(this, arguments));
  }

  (0, _createClass2.default)(StatsSlider, [{
    key: "render",
    value: function render() {
      var props = Object.assign({}, this.props);
      var statCards = [];
      var stats = props.stats,
          className = props.className;
      delete props.stats;
      delete props.className;
      stats.forEach(function (stat, idx) {
        statCards.push(_react.default.createElement("div", {
          key: "stat-".concat(idx),
          className: "stat__card"
        }, stat.dir !== 0 && _react.default.createElement("div", {
          className: "icon ".concat(stat.dir > 0 ? 'up' : '')
        }), _react.default.createElement("div", {
          className: "value"
        }, stat.value), _react.default.createElement("div", {
          className: "title"
        }, stat.title)));
      });
      return _react.default.createElement("div", {
        className: "stat__card__slider ".concat(className || '')
      }, _react.default.createElement("div", {
        className: "stat__card__slider__scroll"
      }, statCards));
    }
  }]);
  return StatsSlider;
}(_react.Component);

exports.default = StatsSlider;
StatsSlider.propTypes = {
  stats: _propTypes.default.array.isRequired
};
StatsSlider.defaultProps = {
  stats: []
};