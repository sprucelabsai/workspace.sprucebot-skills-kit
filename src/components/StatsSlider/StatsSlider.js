import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class StatsSlider extends Component {
	render() {
		let props = Object.assign({}, this.props)
		let statCards = []

		const { stats, className } = props

		delete props.stats
		delete props.className

		stats.forEach((stat, idx) => {
			statCards.push(
				<div key={`stat-${idx}`} className="stat__card">
					{stat.dir !== 0 && (
						<div className={`icon ${stat.dir > 0 ? 'up' : ''}`} />
					)}
					<div className="value">{stat.value}</div>
					<div className="title">{stat.title}</div>
				</div>
			)
		})

		return (
			<div className={`stat__card__slider ${className || ''}`}>
				<div className="stat__card__slider__scroll">{statCards}</div>
			</div>
		)
	}
}

StatsSlider.propTypes = {
	stats: PropTypes.array.isRequired
}

StatsSlider.defaultProps = {}
