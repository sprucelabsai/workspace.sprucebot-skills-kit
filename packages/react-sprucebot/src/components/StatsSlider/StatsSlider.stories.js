import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import StatsSlider from './StatsSlider'
import readme from './StatsSlider.md'

const stories = storiesOf('StatsSlider', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, () => (
		<StatsSlider
			stats={[
				{
					dir: 1,
					value: 12,
					title: 'Things'
				},
				{
					dir: 0,
					value: 3,
					title: 'Stuff'
				},
				{
					dir: 0,
					value: 5,
					title: 'Other Stuff'
				},
				{
					dir: 1,
					value: 12,
					title: 'Things'
				},
				{
					dir: 0,
					value: 3,
					title: 'Stuff'
				},
				{
					dir: 0,
					value: 5,
					title: 'Other Stuff'
				}
			]}
		/>
	))
)
