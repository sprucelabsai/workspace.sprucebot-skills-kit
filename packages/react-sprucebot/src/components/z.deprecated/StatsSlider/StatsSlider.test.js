import React from 'react'
import StatsSlider from './StatsSlider'
import { shallow } from 'enzyme'

describe('StatsSlider tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<StatsSlider />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
