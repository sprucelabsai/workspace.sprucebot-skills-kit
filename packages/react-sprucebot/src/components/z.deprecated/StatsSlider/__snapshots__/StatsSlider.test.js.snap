// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsSlider tests Should match the snapshot 1`] = `
<div
  className="stat__card__slider "
>
  <div
    className="stat__card__slider__scroll"
  />
</div>
`;
